sdk/iot/provisioning/inc/az_iot_provisioning_client.h

```diff
@@ -77,27 +80,37 @@ AZ_NODISCARD az_result az_iot_provisioning_client_init(
  * @brief Gets the MQTT user name.
  *
  * @param[in] client The #az_iot_provisioning_client to use for this call.
- * @param[in] mqtt_user_name An empty #az_span with sufficient capacity to hold the MQTT user name.
- * @param[out] out_mqtt_user_name The output #az_span containing the MQTT user name.
+ * @param[out] mqtt_user_name A buffer with sufficient capacity to hold the MQTT user name.
+ *                            If successful, contains a null-terminated string with the user name
+ *                            that needs to be passed to the MQTT client.
+ * @param[in] mqtt_user_name_size The size, in bytes of #mqtt_user_name.
+ * @param[out] out_mqtt_user_name_length __[nullable]__ Contains the string length in bytes, of 
+ *                                                      #mqtt_user_name. Can be NULL.
  * @return #az_result
  */
-AZ_NODISCARD az_result az_iot_provisioning_client_user_name_get(
+AZ_NODISCARD az_result az_iot_provisioning_client_get_user_name(
     az_iot_provisioning_client const* client,
-    az_span mqtt_user_name,
-    az_span* out_mqtt_user_name);
+    char* mqtt_user_name,
+    size_t mqtt_user_name_size,

```

Comment:
Why aren't these `int32_t`?

Reply:
To interop with C99's libs and `sizeof` and to avoid casting.